### <font color = "Cyan">Hypothesis Statement</font>
<b>Is there a statistically significant relationship between "Shipping Mode", "Region", and the likelihood of "Late Delivery"?

In [1]:
import pandas as pd
df = pd.read_csv("..\Sample Datasets\DataCoSupplyChainDataset1.csv")

In [3]:
selected_cols = ['order_date_(DateOrders)', 'shipping_date_(DateOrders)', 'Shipping_Mode', 'Delivery_Status', 'Days_for_shipping_(real)', 'Days_for_shipment_(scheduled)', 'Late_delivery_risk', 'Sales', 'Sales_per_customer', 'Order_Profit_Per_Order', 'Customer_Segment', 'Order_City', 'Order_Country', 'Order_Region', 'Order_State', 'Order_Status', 'Order_Item_Quantity', 'Order_Item_Discount', 'Order_Item_Discount_Rate', 'Product_Category_Id' , 'Product_Name', 'Product_Price']
refined_data = df[selected_cols]

In [8]:
shipping_mode_summary = refined_data.groupby('Shipping_Mode').agg(
    total_txn = ('Late_delivery_risk', 'count'),
    total_late = ('Late_delivery_risk', 'sum')
)

In [22]:
shipping_mode_summary['Late_percentage'] = (shipping_mode_summary['total_late'] / shipping_mode_summary['total_txn']) * 100
shipping_mode_summary.sort_values(by='Late_percentage', ascending = False)

total_txn  total_late  Late_percentage
Shipping_Mode                                         
First Class         27814       26513        95.322499
Second Class        35216       26987        76.632781
Same Day             9737        4454        45.743042
Standard Class     107752       41023        38.071683

<b>An analysis of the Shipping Performance</b>
* <font color = 'green'>First Class</font> shipments have the highest late delivery rate, with 95.32% of deliveries arriving late.
* <font color = 'green'>Second Class</font> follows with a significantly high late rate of 76.63%.
* <font color = 'green'>Same Day Shipping</font> shows a considerable 45.74% late delivery rate.
* <font color = 'green'>Standard Class</font> performs comparatively better, with 38.07% of its deliveries arriving late.